### PR TITLE
DDF-5709 Removed SAML federated identity code

### PR DESCRIPTION
**Pull request by stustison**
#### Who is reviewing it? 

@SmithJosh @bakejeyner 

#### Select relevant component teams: 
@codice/security 

#### Ask 2 committers to review/merge the PR and tag them here.
@blen-desta
@coyotesqrl
@garrettfreibott
@pklinef
@ryeats 
@rzwiefel
@shaundmorris
@stustison

#### What are the relevant tickets?
Fixes: #5709 

#### Checklist:
- [ ] Documentation Updated
- [ ] Update / Add Threat Dragon models
- [ ] Update / Add Unit Tests
- [ ] Update / Add Integration Tests

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.

### CODE REVIEW

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

File: platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java [hunk 1]
```diff
@@ -92,11 +91,12 @@ public void handleMessage(SoapMessage message) throws Fault {
 
   private synchronized Subject getSubject(String ipAddress) {
     Subject subject = cachedGuestSubjectMap.get(ipAddress);
-    if (Security.getInstance().tokenAboutToExpire(subject)) {
+    if (subject == null) {
       GuestAuthenticationToken token = new GuestAuthenticationToken(ipAddress);
       LOGGER.debug("Getting new Guest user token for {}", ipAddress);
       try {
         subject = securityManager.getSubject(token);
+        // this should be a cache not a map so we can remove items, make this change
```
Comment (blen-desta) [inline platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java]: Do we need to make a ticket for this or track it someway?

File: platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/client/impl/SecureCxfClientFactoryImpl.java [hunk 2]
```diff
@@ -648,11 +642,6 @@ public final T getClientForSystemSubject(Subject subject) {
                   && oauthSecurity != null) {
                 oauthSecurity.setSystemTokenOnClient(
                     WebClient.client(newClient), clientId, clientSecret, discoveryUrl);
-              } else if (!basicAuth
```
Comment (blen-desta) [inline platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/client/impl/SecureCxfClientFactoryImpl.java]: Does this mean we also need to remove the SAML federation option from the sources? Cause the only options look like they are OAuth or Basic.

File: platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java (hunk 1 above)
Comment (stustison) [reply platform/security/interceptor/security-interceptor-guest/src/main/java/org/codice/ddf/security/interceptor/GuestInterceptor.java]: Ah ha! See, I put that in there for myself. Thanks. I need to change that.

File: platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/client/impl/SecureCxfClientFactoryImpl.java (hunk 2 above)
Comment (stustison) [reply platform/security/rest/security-rest-cxfwrapper/src/main/java/org/codice/ddf/cxf/client/impl/SecureCxfClientFactoryImpl.java]: The only SAML thing on sources should now be SAML ECP

**Comment by stustison** [comment]
build now

**Comment by stustison** [comment]
build now

**Pull request merged**